### PR TITLE
Update intro to symbolic

### PR DESCRIPTION
**Pull request by JoelPasvolsky**
Let me know your opinion on whether a last section with a CQM example is desirable, I'm vacillating on that

### CODE REVIEW

File: docs/intro/intro_symbolic_math.rst [hunk 1]
```diff
@@ -4,96 +4,187 @@
...
+  \max_{i,j} \quad ij
 
->>> from dimod import Integer
->>> i = Integer('i')
+  \textrm{s.t.} \quad 2i+2j \le P
+
+where the components are,
+
+* **Variables**: :math:`i` and :math:`j` are the lengths of two sides of the
+  rectangle and and :math:`P` is the length of the perimeter.
```
Comment (arcondello) [inline docs/intro/intro_symbolic_math.rst:20]: `and and`

File: docs/intro/intro_symbolic_math.rst [hunk 2]
```diff
@@ -4,96 +4,187 @@
...
+* **Objective**: maximize the area, which is given by the standard geometric
+  formula :math:`ij`.
+* **Constraint**: subject to (s.t.) not exceeding the given perimeter length;
+  that is, :math:`2i+2j`, the summation of the rectangle's four sides, is
+  constrained to a maximum value of :math:`P`.
+
+*dimod*'s symbolic math enables an intuitive coding of such problems:
+
+>>> print(objective.to_polystring())            # doctest:+SKIP
```
Comment (arcondello) [inline docs/intro/intro_symbolic_math.rst:29]: I had to read this a bunch of times, but I think the first code snippet should be standalone. My first thought is "what is objective?" Something like
```python
>>> i, j = dimod.Binaries(['i', 'j'])
>>> objective = -i*j
>>> constraint = 2*i + 2*j <= 8
```
I think conveys the same point, while also being working code.

On that note, I would adjust the math above to include the word "objective" and "constraint", something like
```tex
\textrm{objective: } \max_{i,j} \quad ij
\textrm{constraint: } \textrm{s.t.} \quad 2i+2j \le 8
```
^ ideally with the alignment fixed, but you get the idea.

Note that I've also used 8 directly. IMO that makes the link super clear between code and math.

File: docs/intro/intro_symbolic_math.rst [hunk 3]
```diff
@@ -4,96 +4,187 @@
...
+.. math::
+
+    \sum_i a_i x_i + \sum_{i \le j} b_{i, j} x_i x_j + c
+
+where :math:`\{ x_i\}_{i=1, \dots, N}` can be integer variables
+(:math:`a_{i}, b_{ij}, c` are real values). If you set :math:`a_1=1` and all
+remaining coefficients to zero, the model represents a single variable,
+:math:`x_1`.
+
+When your variable is used in a linear term of a polynomial, such as :math:`3.7i`,
```
Comment (arcondello) [inline docs/intro/intro_symbolic_math.rst:67]: you swap between `3.7` and `3.75`

File: docs/intro/intro_symbolic_math.rst [hunk 4]
```diff
@@ -4,96 +4,187 @@
...
+the linear bias on the ``'i'``--labeled variable:
+
+>>> 3.75 * i
+QuadraticModel({'i': 3.75}, {}, 0.0, {'i': 'INTEGER'}, dtype='float64')
+
+Similarly, when your variable is in a quadratic term, such as :math:`2.2i^2`, the
+coefficient (:math:`2.2`) is represented in this same model by the value of
+the quadratic bias, :math:`b_{1, 1} = 2.2`:
+
+>>> (3.75 * i + 2.2 * i * i).quadratic
```
Comment (arcondello) [inline docs/intro/intro_symbolic_math.rst:78]: I would just print the whole thing again, for symmetry

File: docs/intro/intro_symbolic_math.rst [hunk 5]
```diff
@@ -4,96 +4,187 @@
...
+where :math:`a_{i}, b_{ij}, c` are real values. The multiplication of two such
+models, with linear terms :math:`a_1 = 1`, reduces to
+:math:`\sum_{i=1} 1 v_1 * \sum_{i=1} 1 u_1 = v_1 u_1`, a multiplication of two
+variables.
+
+In this AND example, because all the variables are the same :class:`~dimod.Vartype`,
+dimod represents each binary variable, and their multiplication, with
+:class:`~dimod.binary.binary_quadratic_model.BinaryQuadraticModel` objects.
+
+>>> bqm_and.vartype == dimod.Vartype.BINARY
```
Comment (arcondello) [inline docs/intro/intro_symbolic_math.rst:125]: `is`

File: docs/intro/intro_symbolic_math.rst [hunk 6]
```diff
@@ -4,96 +4,187 @@
...
+:math:`==`). For example, the constraint of the rectangle problem above,
+
+.. math::
+
+  \textrm{s.t.} \quad 2i+2j \le P
+
+has a ``lhs`` of :math:`2i+2j` less or equal to a ``rhs`` of a some real number
+(:math:`8`):
+
+>>> print(constraint.lhs.to_polystring(), constraint.sense.value, constraint.rhs)  # doctest:+SKIP
```
Comment (arcondello) [inline docs/intro/intro_symbolic_math.rst:169]: why don't I just add a `to_polystring()` method to comparisons? That way you could just do `(2*i + 2*j <= 8).to_polystring()`

File: docs/intro/intro_symbolic_math.rst [hunk 7]
```diff
@@ -4,96 +4,187 @@
...
+
+You can create such an equality or inequality symbolically, and it is shown
+with the model:
+
+>>> print(type(3.75 * i <= 4))
+<class 'dimod.sym.Le'>
+>>> 3.75 * i <= 4
+QuadraticModel({'i': 3.75}, {}, 0.0, {'i': 'INTEGER'}, dtype='float64') <= 4
+
+Performance
```
Comment (arcondello) [inline docs/intro/intro_symbolic_math.rst:180]: There is obviously a lot more to be said here, but I think it makes sense to move that into a future "How to scale your problem" page.

File: docs/intro/intro_symbolic_math.rst (hunk 6 above)
Comment (arcondello) [reply docs/intro/intro_symbolic_math.rst:169]: https://github.com/dwavesystems/dimod/pull/1118

File: docs/intro/intro_symbolic_math.rst (hunk 2 above)
Comment (JoelPasvolsky) [reply docs/intro/intro_symbolic_math.rst:29]: > ```python
> >>> i, j = dimod.Binaries(['i', 'j'])
> ```

Just I used ``i, j = dimod.Integers(["i", "j"])``

File: docs/intro/intro_symbolic_math.rst [hunk 8]
```diff
@@ -4,96 +4,191 @@
...
-x is of type Vartype.BINARY.
-s is of type Vartype.SPIN.
+Consider a simple problem of an AND operation on two binary variables. For
+:math:`\{0, 1\}`--valued binary variables, the AND operation is equivalent to
+multiplication of the two variables:
 
-You can express mathematical functions on these variables using Python functions such
-as :func:`sum`\ [#]_\ :
+>>> x, y = dimod.Binaries(["x", "y"])
+>>> bqm_and = x*y
```
[the remark this answers is not in the document]
Comment (JoelPasvolsky) [reply docs/intro/intro_symbolic_math.rst:98]: I dropped the minus, thanks for catching that!

**Comment by arcondello** [comment]
Sorry, getting back to this, I actually prefer `nand` since you can use binary variables. I think it's easy to add a note distinguishing it from the gate version

**Comment by arcondello** [comment]
I am OK to go with this though if you prefer

**Review by arcondello — APPROVED** [review]

**Pull request merged**